sycl/source/device_selector.cpp

```diff
@@ -51,15 +41,22 @@ static bool isForcedDevice(const device &Dev, int Index = -1) {
   else
     Backend = detail::getSyclObjImpl(Dev)->getPlugin().getBackend();
 
-  for (const detail::device_filter &Filter : FilterList->get()) {
-    if ((Filter.Backend == Backend || Filter.Backend == backend::all) &&
-        (Filter.DeviceType == Type ||
-         Filter.DeviceType == info::device_type::all)) {
-      if (Index < 0 || (Filter.HasDeviceNum && Filter.DeviceNum == Index))
-        return true;
-    }
-  }
-  return false;
+  auto AllDevices = cl::sycl::detail::getSyclObjImpl(Dev.get_platform())
```

Comment:
Please, do not use auto here.

Reply:
i am applied the comment. But why it is not recommended to use auto here?

Reply:
Because the type of AllDevices var is not clear here.